api/src/main/java/org/apache/iceberg/ContentFile.java

```diff
@@ -141,4 +141,13 @@
    *         null value counts, or nan value counts
    */
   F copyWithoutStats();
+
+  /**
+   * Returns the sort order id of this file, which describes how the file is ordered.
+   * This information will be useful for merging data and equality delete files more efficiently
+   * when they share the same sort order id.
+   */
+  default Integer sortOrderId() {
+    return null;
```

Comment:
Could you add this after `equalityFieldIds` instead of here? The copy methods should be last because they aren't fields in the file.